R/ancombc_prep.R

```diff
@@ -68,7 +76,15 @@ tse_construct = function(data, assay_name, tax_level, phyloseq) {
             tax_level = "ASV"
             tse_alt = tse
         } else {
-            tse_alt = mia::agglomerateByRank(tse, tax_level)
+            # Check if tax_level parameter belongs to taxonomyRanks
+            if (is.character(tax_level) && length(tax_level) == 1 && tax_level %in% mia::taxonomyRanks(tse)) {
+                #Merge using agglomerateByRank
+                tse_alt <- mia::agglomerateByRank(tse, tax_level)
+            } else {
+                # Merge using mergeRows
+                tse_alt <- mia::mergeRows(tse, tax_level)
+            }
+            tse_alt
```

Comment:
same

Reply:
Should we export mia::.merge_features?

Reply:
No but I think you can it with triple semicolon. Check with TB

Reply:
notably mia:::.merge_features is available from mia 1.9.9

Reply:
As pointed out by @Daenarys8 that the `mia:::.merge_features` function becomes available starting from `mia` version 1.9.9, while the currently released version of `mia` on Bioconductor remains at 1.8.0. Consequently, the ".merge_features" function is currently not accessible.

```R
> mia:::.merge_features
Error: object '.merge_features' not found
```

I'm wondering the best approach to import this function. One option is to consider copying the function directly from [this source](https://github.com/microbiome/mia/blob/master/R/utils.R) into the `ANCOMBC` utils functions. I will update the package accordingly in the future once this function becomes available in the release version of mia. Any ideas? @Daenarys8 @antagomir 

Thank you,
Huang